docs/getting_started/typescript/hello_world_in_typescript/index.md

```diff
@@ -88,142 +158,155 @@ Create the `package.json` file in the root of your project and add the following
-export async function example(name: string): Promise<string> {
-  return await greet(name);
+```ts
+// Use the IP address to get the location.
+export async function getLocationInfo(ip: string): Promise<string> {
+  const url = `http://ip-api.com/json/${ip}`;
+  const response = await fetch(url);
+  const data = await response.json();
+  return `${data.city}, ${data.regionName}, ${data.country}`;
 }
 ```
-<!--SNIPEND-->
 
-In this code, the variable `greet` is assigned the value of `proxyActivites`, which is a method from the Temporal TypeScript SDK that lets you configure the Activity with different options. In this example, you have specified that the Start-to-Close Timeout for your Activity will be one minute, meaning that your Activity has one minute to begin before it times out. Of all the Temporal timeout options, `startToCloseTimeOut` is the one you should always set.
+This Activity follows the same pattern as the `getIP` Activity. It's an exported async function that uses `fetch` to call a remote service. This time, the service returns JSON data rather than text.
 
-The `example` function executes an Activity called `greet`, and the function returns the result of the Activity. You'll create the `greet` function in the next section.
+While Activities can accept input argument, it's a best practice to send a single argument rather than multiple arguments. In this case you only have a single String. If you had more than one argument, you should bundle them up in a serializable object.  Review the [Activity parameters](https://docs.temporal.io/dev-guide/typescript/foundations/#activity-parameters) section of the Temporal documentation for more details, as there are some limitations you'll want to be aware of when running more complex applications.
 
-:::tip
-No matter how many inputs you pass to a Workflow, it's a best practice to use an object. There are some JSON restrictions that you should be aware of, and you can learn more about the kinds of inputs you can pass to a Workflow in the [Workflow parameters](https://docs.temporal.io/dev-guide/typescript/foundations#workflow-parameters) section of the Temporal documentation.
+You've created your two Activities. Now you'll coordinate them using a Temporal Workflow.
 
-:::
+## Control application logic with a Workflow
 
-Next, you'll define the Activity that your Workflow will execute.
+Workflows are where you configure and organize the execution of Activities. You define a Workflow by writing a *Workflow Definition* using one of the Temporal SDKs. Review the [Develop Workflows](https://docs.temporal.io/develop/typescript/core-application#develop-workflows) section of the Temporal documentation for more about Workflows in TypeScript.
 
-## Create an Activity
+Open the file `src/workflows.src` and remove the code in the file since you'll add your own. Then add the following code to import the Activities and configure how the Workflow should handle failures:
```

Comment:
I would add a link to the different retry policies since there isn't too much of an explanation of what they are for.

Consider something like this:

```suggestion
Open the file `src/workflows.src` and remove the code in the file since you'll add your own. Then add the following code to import the Activities and configure how the Workflow should handle failures with a [retry policy](https://docs.temporal.io/encyclopedia/retry-policies).
```